src/librustc_typeck/astconv.rs

```diff
@@ -696,7 +696,12 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                 }
                 (GenericParamDefKind::Type { has_default, .. }, GenericArg::Type(ty)) => {
                     if *has_default {
-                        tcx.check_stability(param.def_id, Some(arg.id()), arg.span());
+                        tcx.check_stability_internal(
+                            param.def_id,
+                            Some(arg.id()),
+                            arg.span(),
+                            false,
```

Comment:
Why do we have to override deprecation checking for generic parameters?

Reply:
If the parent is deprecated, the user will already have a warning, If we were to produce another warning we would see:
```
error: use of deprecated item 'intrinsics::init::T': superseded by MaybeUninit, removal planned
 |     pub fn init<T>() -> T;
```

Reply:
Can you add a comment to this effect?

Reply:
Will do